tests/cephfs/clients/ceph_auth_caps_modifying_permissions.py

```diff
@@ -0,0 +1,133 @@
+            for perm1, perm2 in combs_perm:
+                if daemon1 == "mds" and "x" in perm1:
+                    break
+                elif daemon2 == "mds" and "x" in perm2:
+                    break
+                else:
+                    client1.exec_command(
+                        sudo=True,
+                        cmd=f"ceph auth caps client.{rand_name} {daemon1} {perm1} {daemon2} {perm2}",
+                    )
+                    out2, _ = client1.exec_command(
+                        sudo=True,
+                        cmd=f"ceph auth get client.{rand_name} -f json-pretty",
+                    )
+                    output2 = json.loads(out2)[0]
+                    if output2["caps"][daemon1] != perm1.strip("'"):
+                        raise CommandFailed(
+                            f"Not Expected permission for {daemon1}, it should be {perm1}"
+                        )
+                    if output2["caps"][daemon2] != perm2.strip("'"):
+                        raise CommandFailed(
+                            f"Not Expected permission for {daemon2}, it should be {perm2}"
+                        )
+
+        combs_daemon = list(itertools.combinations(daemon_list, 3))
+        for daemon1, daemon2, daemon3 in combs_daemon:
+            combs_perm = list(itertools.combinations(permission_list, 3))
+            for perm1, perm2, perm3 in combs_perm:
+                print(combs_daemon)
+                print(combs_perm)
+                if daemon1 == "mds" and "x" in perm1:
+                    break
+                elif daemon2 == "mds" and "x" in perm2:
+                    break
+                elif daemon3 == "mds" and "x" in perm3:
+                    break
+                else:
+                    client1.exec_command(
+                        sudo=True,
+                        cmd=f"ceph auth caps client.{rand_name} {daemon1} {perm1} {daemon2} {perm2} {daemon3} {perm3}",
+                    )
+                    out3, _ = client1.exec_command(
+                        sudo=True,
+                        cmd=f"ceph auth get client.{rand_name} -f json-pretty",
+                    )
+                    output3 = json.loads(out3)[0]
+                    print((daemon1, daemon2, daemon3), (perm1, perm2, perm3))
+                    if output3["caps"][daemon1] != perm1.strip("'"):
+                        raise CommandFailed(
+                            f"Not Expected permission for {daemon1}, it should be {perm1}"
+                        )
+                    if output3["caps"][daemon2] != perm2.strip("'"):
+                        raise CommandFailed(
+                            f"Not Expected permission for {daemon2}, it should be {perm2}"
+                        )
+                    if output3["caps"][daemon3] != perm3.strip("'"):
+                        raise CommandFailed(
+                            f"Not Expected permission for {daemon3}, it should be {perm3}"
```

Comment:
Along with this, can we also do some IO operations based on the Permission set.
we can mount the cephfs using the user and do IO operations

Reply:
I have added IO operation at the beginning permission modifying.

Reply:
@julpark-rh - After every permission change, please run IO's and validate the permissions accordingly.